example/review/src/main/scala/review/ReviewEnv.scala

```diff
@@ -0,0 +1,32 @@
+package review
+
+trait ReviewEnv { def service: ReviewEnv.Service }
+
+object ReviewEnv {
+
+  trait Service {
+
+    def getReviews: List[Review]
+    def getReview(id: Int): Option[Review]
+  }
+
+  val inMemory: ReviewEnv = new ReviewEnv {
+    override def service: Service = new Service {
+
+      val reviews = Review(
+        id = 0,
+        key = Some("review 0"),
+        State(id = 0)
+      ) :: Review(
+        id = 1,
+        key = Some("review 1"),
+        State(id = 0)
+      ) :: Nil
+
+      def getReviews: List[Review] = reviews
+
+      def getReview(id: Int): Option[Review] =
+        reviews.find(_.id == id)
+    }
+  }
+}
```

Comment:
please keep it as simple as possible and don't use patterns coming from ZIO for example that are used there for their internal reasons.

```suggestion
trait ReviewService {
  def getReviews: List[Review]
  def getReview(id: Int): Option[Review]
}

object ReviewService{
  val inMemory: ReviewService = new ReviewService {
    val getReviews: List[Review] = Review(
      id = 0,
      key = Some("review 0"),
      State(id = 0)
    ) :: Review(
      id = 1,
      key = Some("review 1"),
      State(id = 0)
    ) :: Nil

    def getReview(id: Int): Option[Review] =
      reviews.find(_.id == id)
  }
}
```

Reply:
🙏 I made all the changes [here](https://github.com/sangria-graphql/sangria-federated/pull/6/commits/5779a691415fb2ae58c95b19dd0bb11b0bc0c0ca)